### DISCUSSION TITLE
post/azure-functions-identity-based-connections

### DISCUSSION THREAD

**Discussion by giscus** [markheath/soundcode-blog-comments#81] [Announcements]
# post/azure-functions-identity-based-connections

Mark Heath's Development Blog

https://www.markheath.net/post/azure-functions-identity-based-connections

**Comment by antonburger-alcove** [comment]
Thanks for this! In function apps using libraries for various Azure services (directly, in function code, rather than in triggers and bindings) I had already started doing this by assigning managed identities to function apps and putting only resource URLs in app settings - then using `DefaultAzureCredential` in code to use the assigned identity when instantiating SDK clients - but I had no idea it was supported for trigger/binding extensions as well. Not sure how I missed that in the docs!

**Comment by webbsk** [comment]
This is a really great article. Can I just make the suggestion that you should also share what principalId you used to assign the roles to for the storage account and how you found it? I assumed you looked up the principalId from the system assigned identity for the azure function but I'm not sure (a) if that's intended (b) if beginner's could figure that out. Otherwise, thank you for sharing because it's much appreciated!

**Reply by markheath** [reply]
A good catch, sorry I somehow missed that step in the post. I'm pretty sure this is what I meant to put in, so will get the post updated!

```powershell
$principalId = az functionapp identity show -n $functionAppName `
  -g $resourceGroupName --query principalId -o tsv
```
